### PR TITLE
feat: add first steps - install, hello world

### PR DESCRIPTION
**Pull request by TaricsArpi**
- Added first steps guide for:
  - installation with asdf
  - hello world

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,86 @@
-# elixir-starting-guide
\ No newline at end of file
+# How to install Elixir with asdf?
+
+## Why is it important?
+- Elixir runs on the Erlang VM
+- If you install less-compatible versions of Erlang and Elixir you might
```
Comment (Shadowbeetle) [inline README.md]: If you install incompatible

File: README.md [hunk 2]
```diff
@@ -1 +1,86 @@
...
+run into problems when trying to run your code
+- The version of Erlang is often referred as the OTP version. The version of Elixir used must be compiled specifically for the version of Erlang in use.
```
Comment (Shadowbeetle) [inline README.md]: The version of Erlang is often referred as the OTP version. You can go a long way with Elixir without understanding what OTP is. Just remember that Erlang/OTP is often used a bit similarly to GNU/Linux and its relation to Erlang is also similar to GNU's relation to Linux.

File: README.md [hunk 3]
```diff
@@ -1 +1,86 @@
...
+
+!["elixir -v"](/images/installation/elixir-v.png)
+
+- Note the first line Erlang/OTP 26 must match the Erlang version that Elixir was compiled against. If they do not match, you may encounter issues.
```
Comment (Shadowbeetle) [inline README.md]: It's best to use the version of Erlang Elixir was compiled against. You can use `elixir -v` to figure out which version you need, as highlighted above.

File: README.md [hunk 4]
```diff
@@ -1 +1,86 @@
...
+- Knowing you can use `elixir -v` to see both versions helps you quickly tell when you have compatible versions being used together.
```
Comment (Shadowbeetle) [inline README.md]: This can be removed if my previous comment is applied

File: README.md [hunk 5]
```diff
@@ -1 +1,86 @@
...
+- :warning: **Please exercise caution when selecting the version. It should be compatible with the Erlang version you have installed before!** :warning:
+- In our case, let's pick the latest stable version built with the previously installed Erlang version: `1.16.0-otp-26`
+
+`asdf install elixir 1.16.0-otp-26`
+
+- After the installation check your Elixir version and you should see the following:
+
+!["elixir-v"](/images/installation/elixir-v.png)
+
+## Activating Elixir and Erlang versions
```
Comment (Shadowbeetle) [inline README.md:69]: Setting versions

File: README.md [hunk 6]
```diff
@@ -1 +1,86 @@
...
+- Although we installed the versions we want, they aren’t yet activated. There are two ways to activate a version.
```
Comment (Shadowbeetle) [inline README.md]: aren't yet set.

File: src/hello_world.exs [hunk 7]
```diff
@@ -0,0 +1 @@
+IO.puts("Hello World!")
```
Comment (Shadowbeetle) [inline src/hello_world.exs:1]: Where to go from here?

**Review by Shadowbeetle — CHANGES_REQUESTED** [review]
I left some comments. The basics are good, but we'll need to work on that a bit more.

File: README.md [hunk 8]
```diff
@@ -1 +1,86 @@
-# elixir-starting-guide
\ No newline at end of file
+# How to install Elixir with asdf?
+
+## Why is it important?
+- Elixir runs on the Erlang VM
```
Comment (TaricsArpi) [inline README.md]: Comparison with JS/TS, how it compiles to bitcode

File: README.md [hunk 9]
```diff
@@ -1 +1,86 @@
...
+!["elixir -v"](/images/installation/elixir-v.png)
+
+- Note the first line Erlang/OTP 26 must match the Erlang version that Elixir was compiled against. If they do not match, you may encounter issues.
+- Knowing you can use `elixir -v` to see both versions helps you quickly tell when you have compatible versions being used together.
+
+## Why use asdf?
+- asdf is CLI tool which can manage multiple language runtimes - so you
+can install two matching versions of Erlang and Elixir to avoid the above
+mentioned problems
+
```
Comment (TaricsArpi) [inline README.md:21]: What is "asdf"? Is it worth to use asdf for node as well?

File: README.md [hunk 10]
```diff
@@ -1 +1,86 @@
...
+## Install Elixir
+- For Elixir we need to install the asdf plugin first again:
+
+`asdf plugin-add elixir https://github.com/asdf-vm/asdf-elixir.git`
+
+- List the available versions with `asdf list-all elixir` and pick the desired version
+
+!["elixir-versions"](/images/installation/elixir-list-all.png)
+
+- :warning: **Please exercise caution when selecting the version. It should be compatible with the Erlang version you have installed before!** :warning:
```
Comment (TaricsArpi) [inline README.md:58]: "This is different compared to JS"....

File: README.md [hunk 11]
```diff
@@ -1 +1,86 @@
...
+`asdf install elixir 1.16.0-otp-26`
+
+- After the installation check your Elixir version and you should see the following:
+
+!["elixir-v"](/images/installation/elixir-v.png)
+
+## Activating Elixir and Erlang versions
+- Although we installed the versions we want, they aren’t yet activated. There are two ways to activate a version.
+    - local
+    - global
```
Comment (TaricsArpi) [inline README.md]: Does "nvm" has something similar for local / global version usage?

File: README.md [hunk 12]
```diff
@@ -1 +1,86 @@
...
+- asdf is CLI tool which can manage multiple language runtimes - so you
+can install two matching versions of Erlang and Elixir to avoid the above
+mentioned problems
+
+## How to install asdf?
+- Follow the instructions [here]!(https://asdf-vm.com/guide/getting-started.html#_3-install-asdf)
+
+## Install Erlang
+- First we need to install the asdf plugin that installs Erlang and manages the versions for us.
+- The following command adds the Erlang plugin:
```
Comment (TaricsArpi) [inline README.md:31]: We don't have something like this in nvm because it only handles node, that's why we need these plugins.

File: README.md [hunk 13]
```diff
@@ -1 +1,86 @@
...
+
+# Hello World in Elixir
+
+- Let's write our first very simple program in Elixir, the Hello World.
+- First launch an IEx (Interactive Elixir) in the terminal:
+
+!["iex-launch"](/images/hello_world/iex.png)
+
+- Now you can run Elixir code in your terminal.
+- Let's just type in `IO.puts("Hello World!")` to the terminal:
```
Comment (TaricsArpi) [inline README.md]: Compare this with JSs "console.log"
IEx command is like "node"

File: README.md [hunk 14]
```diff
@@ -1 +1,86 @@
...
+
+!["elixir-hello-world-terminal"](/images/hello_world/elixir-hello-world-terminal.png)
+
+- IEx will print the message and then display the return value of a function, which is an atom `:ok`(indicating that the function exectued successfully)
+
+- Alternatively we can put this one line of code to its own file. Let's call it `hello_world.exs`:
+
+!["elixir-hello-world-file"](/images/hello_world/elixir-hello-world-file.png)
+
+- We can execute this script like this:
```
Comment (TaricsArpi) [inline README.md:111]: Add compiled version as well

File: README.md [hunk 15]
```diff
@@ -1 +1,86 @@
...
+mentioned problems
+
+## How to install asdf?
+- Follow the instructions [here]!(https://asdf-vm.com/guide/getting-started.html#_3-install-asdf)
+
+## Install Erlang
+- First we need to install the asdf plugin that installs Erlang and manages the versions for us.
+- The following command adds the Erlang plugin:
+
+`asdf plugin-add erlang https://github.com/asdf-vm/asdf-erlang.git`
```
Comment (Shadowbeetle) [inline README.md]:
```bash
$ asdf plugin-add erlang https://github.com/asdf-vm/asdf-erlang.git
```

**Review by Shadowbeetle — CHANGES_REQUESTED** [review]

File: README.md [hunk 16]
```diff
@@ -1 +1,106 @@
...
+- Elixir has gained popularity, particularly in the context of developing robust and scalable web applications using the Phoenix framework. 
+- Elixir differs from JavaScript in several key aspects. While JavaScript is primarily a front-end language often used for client-side web development, Elixir is a back-end language designed for concurrent, distributed systems.
+- Elixir leverages functional programming principles, immutability, and pattern matching, offering a different paradigm compared to JavaScript's more object-oriented and imperative approach.
+- Elixir's robust concurrency model, inspired by Erlang's actor-based system, enables seamless scalability and fault tolerance, making it particularly suitable for building reliable and high-performance server-side applications, a domain where JavaScript, traditionally, has been less focused.
```
Comment (Shadowbeetle) [inline README.md:9]: This smells like SEO :D can you give a more personal description from your point of view? Keep this on for now, and add your thoughts as well below in a separate section and we'll merge the two

**Review by Shadowbeetle — CHANGES_REQUESTED** [review]

**Pull request merged**